OmniKit/OmnipodCommon/MessageBlocks/DetailedStatus.swift

```diff
@@ -220,3 +220,23 @@ public struct ErrorEventInfo: CustomStringConvertible, Equatable {
         self.podProgressStatus = PodProgressStatus(rawValue: rawValue & 0xF)!
     }
 }
+
+//
+// Convenience functions for dealing with ReserviorLevel readings.
+// Historically these were optionals with no reading representing 50+.
+//
+public func isValidReservoirLevelValue(_ reservoirLevel: Double?) -> Bool
```

Comment:
I would prefer that debug representations of DetailedStatus and StatusResponse keep the original value, or a close representation of it. These structs should stay closer to represent the protocol level info.  If we got some other value that was not <= maximumReservoirReading, and not the 50+ level, the debug representation should show it.  And generally, we shouldn't be exposing public global functions. If you wanted to do something like this, it should be on the DetailedStatus and StatusResponse structs, even if that means duplicated code.

Reply:
Currently the DetailedStatus and StatusResponse are inconsistent with each other in their display of their reservoirLevel’s value when they should be the same. The StatusResponse uses “\(String(describing: reservoirLevel))” and DetailedStatus uses “\(reservoirLevel > Pod.maximumReservoirReading ? "50+" : reservoirLevel.twoDecimals) U”. After looking at a lot of decoded Omnipod communications, I found that I preferred the DetailedStatus style for both.

I understand the concern to see the actual raw values in the case that there ever might a reservoir value above 50.0 which wasn’t exactly 51.15, but this 51.15U value isn’t really that special (it’s just the result of the particular 11-bit 0x3FF value used by the pods for this condition) as any value greater than 50.0U would need to be treated as 50+U by the controlling software anyways. So I believe that showing the actual raw value isn’t really useful for the particular case of the Omnipod reservoir value.

I did find that using utility functions for dealing with testing whether the reservoir level is valid and for generating a uniform string to display for its value was useful both for sharing more OmnipodCommon code between the various implementations (e.g., Loop 2 which uses an optional value and Loop 3 which does not) and so that other places in Omni{BLE,Kit}{,UI} don’t have to know the details of how to generate a properly generated reservoir value string so even more code can be the same between various implementations (N.B., I didn’t update other places in Omni{BLE,Kit} that could use these functions in this particular PR). The only reason why I needed to include public on these utility func’s was because OmniKitUI wouldn’t be able to access them otherwise (but not so for OmniBLE UI for some reason) which was unfortunate.

Reply:
Decided to remove the isValidReservoirLevelValue() and reservoirLevelString() funcs and have both DetailedStatus and StatusResponse consistently use “50+” only for the exact 0x377 reservoir value using a duplicated inline ternary conditional operator.